refactor(linter): pass paths to `TsGoLintState.lint` method

Sysix opened the pull request:
The language server does need to hold all paths. It will lint when the client is requesting a file

camc314 commented:
### Merge activity

* **Aug 15, 10:14 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(linter): pass paths to `TsGoLintState.lint` method (#13131)

The language server does need to hold all paths. It will lint when the client is requesting a file
